zerver/tests/test_realm_export.py

```diff
@@ -0,0 +1,77 @@
+from mock import patch
+
+from django.test import override_settings
+from django.conf import settings
+from django.utils.timezone import now as timezone_now
+
+from zerver.lib.test_classes import ZulipTestCase
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.test_helpers import use_s3_backend, create_s3_buckets
+from zerver.views.public_export import public_only_realm_export
+from zerver.models import RealmAuditLog
+
+class RealmExportTest(ZulipTestCase):
+    @use_s3_backend
+    def test_hit_export_realm_endpoint(self) -> None:
+        admin = self.example_user('iago')
+        self.login(admin.email)
+        create_s3_buckets(
+            settings.S3_AUTH_UPLOADS_BUCKET,
+            settings.S3_AVATAR_BUCKET)
+
+        with patch('zerver.lib.export.do_export_realm'):
+            with patch('zerver.views.public_export.queue_json_publish') as mock_publish:
+                result = self.client_post('/json/export/realm')
+                queue_data = mock_publish.call_args_list[0][0]
+                worker = mock_publish.call_args_list[0][0][0]
+        self.assert_json_success(result)
+        mock_publish.assert_called_once()
+
+        event = queue_data[1]
+        self.assertEqual(worker, 'deferred_work')
+        self.assertEqual(event['realm_id'], 1)
+        self.assertEqual(event['type'], 'realm_exported')
+        self.assertEqual(event['realm_name'], 'zulip')
+
+        with patch('zerver.lib.export.do_export_realm') as mock_export:
+            result = self.client_post('/json/export/realm')
+            args = mock_export.call_args_list[0][1]
+        self.assert_json_success(result)
+
+        self.assertEqual(args['realm'], admin.realm)
+        self.assertEqual(args['public_only'], True)
+        self.assertEqual(args['output_dir'].startswith('/tmp/zulip-export-'), True)
+        self.assertEqual(args['threads'], 6)
+
+        with patch('zerver.lib.export.do_export_realm'):
+            with override_settings(LOCAL_UPLOADS_DIR='/var/uploads'):
+                result = self.client_post('/json/export/realm')
+        self.assert_json_success(result)
```

Comment:
I'm wondering if a simple success assertion is enough here, or if I should check the `args_list` as well.

Reply:
Yes, we should check `args_list` here as well.  As a general rule, when you're mocking a function, you generally want some assertion about the function being called correctly, and this is the right choice for verifying this code.